vidaud/seta
# [seta-react] Add the new search input

## Description

This PR introduces the following changes:

- Added a new page and route - `/search-new` to run side-by-side with the original one until it's feature complete
- For now it supports the whole range of autocompletions - suggestions, related clusters, related terms
- Introduced `react-query` for optimized API calls together with the associated lining rules
- Added Tailwind CSS for its utility classes with the goal of eventually replacing Prime Flex

Also added a temporary link in the menu for `Search New` to make it easier to access the new page.

## How to test

- Rebuild the React container to install the new packages
- Go to http://localhost/search-new or click on the `Search New [WIP]` menu link
- Check the functionality of the new search input and autocompletion
- Check that the "loading" and "no data" states are properly displayed for all the suggestions
- Check that the "error" state allows retrying the failing query - this can be simulated by turning off the internet connection or setting the throttling option to `Offline` in the `Network` tab, and selecting an uncached token
- Verify that the API calls are properly cached and don't get called more than necessary
- Verify that the UI behaves similarly on both Chrome and Firefox
